superset/security/manager.py

```diff
@@ -1054,19 +1054,24 @@ def raise_for_access(
             denied = set()
 
             for table_ in tables:
-                schema_perm = self.get_schema_perm(database, schema=table_.schema)
```

Comment:
`table._schema` could be the one defined in the query itself, or the one selected on the left dropdown.
For that reason, we could be fetching the permission for a schema that does not contain the table, and validate the request (if the user has permission to that schema).
So, we invert the request order here, by querying all the datasources for the database/table combination, and then matching the schema. If there's a match, then we check the permission on that schema.